### DISCUSSION TITLE
PHP 8.1 slow especially with images

### DISCUSSION THREAD

**Discussion by bartosg-acg** [dompdf/dompdf#3490] [General]
Hi.
I tried it on PHP 8.1 and it is extremly slow especially with images and with DejaVu font.
Php 7.4 is working good.
Have any settings or solution for this?

**Comment by bsweeney** [comment]
The presence of XDebug can impact performance. Other than that, no general recommendations.

Which version of Dompdf are you using? Have you checked to see if there's excessive PHP logging?

**Comment by bartosg-acg** [comment]
8.1 without xdebug.

compser says version is:
"dompdf/dompdf": "^3.0"

**Comment by bsweeney** [comment]
Can you provide a sample of the HTML / CSS and images? All things being equal there _could_ be a slight performance hit due to the addition of character font matching, but we didn't see significant impact in the testing we performed.

**Comment by bartosg-acg** [comment]
Here in the 
[test.zip](https://github.com/user-attachments/files/16086842/test.zip)

The html base meta you need to change

**Reply by bsweeney** [reply]
I'm not seeing any issues with the supplied document. Are there any rendering issues, or is it just slower?

You could try [enabling additional debug messaging](https://github.com/dompdf/dompdf/wiki/Troubleshooting-Dompdf#configuration) to see if anything relevant is logged.
